gjs: 1.64.2 -> 1.64.3

r-ryantm opened the pull request:
Semi-automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/gjs/versions.


meta.description for gjs is: "JavaScript bindings for GNOME"

meta.homepage for gjs is: "https://gitlab.gnome.org/GNOME/gjs/blob/master/doc/Home.md"

meta.changelog for gjs is: ""


###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- gjs result is not automatically checked, because some tests take a long time to run

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
130 total rebuild path(s)

44 package rebuild(s)

44 x86_64-linux rebuild(s)
43 i686-linux rebuild(s)
1 x86_64-darwin rebuild(s)
42 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
adapta-gtk-theme
chrome-gnome-shell
discover
flatpak
flatpak-builder
gjs
gnome-builder
gnome3.anjuta
gnome3.gjs
gnome3.gnome-books
gnome3.gnome-characters
gnome3.gnome-control-center
gnome3.gnome-documents
gnome3.gnome-maps
gnome3.gnome-session
gnome3.gnome-shell
gnome3.gnome-software
gnome3.gnome-sound-recorder
gnome3.gnome-terminal
gnome3.gnome-tweak-tool
gnome3.gnome-tweaks
gnome3.gnome-weather
gnome3.gnome_control_center
gnome3.gnome_session
gnome3.gnome_shell
gnome3.gnome_terminal
gnome3.gpaste
gnome3.polari
gnome3.pomodoro
gnome3.sushi
gnomeExtensions.drop-down-terminal
gnomeExtensions.gsconnect
gtkpod
haskellPackages.gi-ostree
malcontent-ui
minishift
ostree
pantheon.appcenter
pantheon.elementary-session-settings
pantheon.sideload
plasma5.discover
rpm-ostree
xdg-desktop-portal
xdg-desktop-portal-gtk
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/k04226b2sihszn27zdqca0jmhf3d1rp2-gjs-1.64.3 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A gjs https://github.com/r-ryantm/nixpkgs/archive/2df0fef39ea5278d6fcd8ec0d09e6f51c9f1666c.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/k04226b2sihszn27zdqca0jmhf3d1rp2-gjs-1.64.3
ls -la /nix/store/k04226b2sihszn27zdqca0jmhf3d1rp2-gjs-1.64.3/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
<br>- gnomeExtensions.icon-hider
<br>- gnomeExtensions.system-monitor
<br>- osxfuse
<br>- osxsnarf
</details>
<details>
  <summary>2 packages failed to build:</summary>
<br>- pantheon.appcenter
<br>- rpm-ostree
</details>
<details>
  <summary>34 packages built:</summary>
<br>- adapta-gtk-theme
<br>- chrome-gnome-shell
<br>- discover
<br>- flatpak
<br>- flatpak-builder
<br>- gjs
<br>- gnome-builder
<br>- gnome3.anjuta
<br>- gnome3.gnome-books
<br>- gnome3.gnome-characters
<br>- gnome3.gnome-control-center
<br>- gnome3.gnome-documents
<br>- gnome3.gnome-maps
<br>- gnome3.gnome-session
<br>- gnome3.gnome-shell
<br>- gnome3.gnome-software
<br>- gnome3.gnome-sound-recorder
<br>- gnome3.gnome-terminal
<br>- gnome3.gnome-tweak-tool
<br>- gnome3.gnome-weather
<br>- gnome3.gpaste
<br>- gnome3.polari
<br>- gnome3.pomodoro
<br>- gnome3.sushi
<br>- gnomeExtensions.drop-down-terminal
<br>- gnomeExtensions.gsconnect
<br>- gtkpod
<br>- malcontent-ui
<br>- minishift
<br>- ostree
<br>- pantheon.elementary-session-settings
<br>- pantheon.sideload
<br>- xdg-desktop-portal
<br>- xdg-desktop-portal-gtk
</details>

---

###### Maintainer pings

cc @hedning @jtojnar @worldofpeace for testing.

r-ryantm committed:
gjs: 1.64.2 -> 1.64.3

jtojnar reviewed (approved):
Changes okay: https://ftp.gnome.org/pub/GNOME/sources/gjs/1.64/gjs-1.64.3.news

jtojnar reviewed (changes requested):
Installed tests fail.

jtojnar committed:
gjs: Fix tests

Installed tests require GSettings schemas.

And the build-time tests are now passing.

jtojnar committed:
gjs: Fix locale detection

The build was complaining:

	meson.build:660: WARNING: Your libc does not have the C.UTF-8 locale and no other
	suitable UTF-8 fallback locale could be found. You can still build GJS, but
	some tests will fail.

The tests do not actually seem to be affected but still.

jtojnar commented:
This was still awaiting review.

ryantm commented:
There was an accidental mass closing of bot prs sorry.

jtojnar commented:
@GrahamcOfBorg build gjs
@GrahamcOfBorg test installedTests.gjs
